File: modules/MRInVariousLanguages/source/VariousLanguages/MR.py
def mapper(key, value):
    words=key.split()
    for word in words:
        Wmr.emit(word, '1')

def reducer(key, iter):
    sum = 0
    for s in iter:
        sum = sum + int(s)
    Wmr.emit(key, str(sum))

File: modules/MRInVariousLanguages/source/VariousLanguages/MR.java
/* Mapper for word count */

class Mapper {
  public void mapper(String key, String value) {
    String words[] = key.split(" ");
    int i = 0;
    for (i = 0;  i < words.length;  i++)
      Wmr.emit(words[i], "1");
  }

}

/* Reducer for word count */

class Reducer {
  public void reducer(String key, WmrIterator iter) {
    int sum = 0;
    while (iter.hasNext()) {
      sum += Integer.parseInt(iter.next());
    }
    Wmr.emit(key, Integer.valueOf(sum).toString());
  }

}


